community/learn/graphql-tutorials/tutorials/react-apollo-hooks/app-final/src/components/Todo/TodoPublicList.js

```diff
@@ -0,0 +1,132 @@
+import React, { Fragment, useState, useEffect } from 'react';
+import { useSubscription, useApolloClient } from '@apollo/react-hooks';
+import gql from 'graphql-tag';
+import TaskItem from './TaskItem';
+
+const TodoPublicList = (props) => {
+  const [state, setState] = useState({
+    olderTodosAvailable: props.latestTodo ? true : false,
+    newTodosCount: 0,
+    error: false,
+    todos: [],
+    oldestTodoId: props.latestTodo ? (props.latestTodo.id + 1) : 0,
```

Comment:
OldestTodoId can be moved out of the state but newestTodoId can't be since if it changes according to the new props, the if condition in the useEffect would never be true and hence new todo as notification button won't come. Let me know what should be done.

Reply:
`newestTodoId` is inferred from the `todos` array. You already have `todos` in the state. You can definitely declare a local variable that infers the required values.

If it's too much effort, you can skip this. But I think this is the core part of the subscriptions logic and it's better if it is clean.

You make the call :)

Reply:
Yeah makes sense, I will do that. I forgot that todos anyway can be used, my bad.

Reply:
Done! Certainly better now, thanks for suggesting this change.